### ISSUE TITLE
Octave compatibility: check for empty structures before calling cat

### ISSUE THREAD

**Issue by kupiqu** [DynaSim/DynaSim#222]
```
error: concatenation operator not implemented for 'scalar' by 'scalar struct' operations
error: called from
    dsCombineModels at line 36 column 19
    dsGenerateModel at line 238 column 10
    dsCheckModel at line 98 column 8
    dsSimulate at line 484 column 7
    dsDemos at line 87 column 5
```

**Comment by kupiqu** [comment]
The problem is that (empty) structures are not properly initialized, so when they are empty they are in fact scalars, and Octave doesn't like that.

For this specific example, I can workaround the issue with:

```
if isempty(model1.conditionals)
  model1.conditionals = struct ('');
end
```

However I think a proper fix is to properly initialize structures as such when they may not be filled with any content.

@jsherfey as this involves other parts of the code, I think it's better if you properly fix this...

**Comment by kupiqu** [comment]
Same issue applies to model.linkers (line 36 in dsCombineModels.m file)

**Comment by jsherfey** [comment]
The proper place to initialize fields of the model structure is in dsCheckmodel(). We might need to do something similar elsewhere, but that's a start. I'll work on that.

**Comment by jsherfey** [comment]
Solved in 266ae0a36c638aa942caf02053cedf00e14f32c5
